src/Shouldly/ShouldlyExtensionMethods/ShouldBe/ShouldBeDateTimeTestExtensions.cs

```diff
@@ -12,29 +12,89 @@ public static void ShouldBe(this DateTime actual, DateTime expected, TimeSpan to
             actual.AssertAwesomely(v => Is.Equal(v, expected, tolerance), actual, expected, tolerance);
         }
 
+        public static void ShouldBe(this DateTime actual, DateTime expected, TimeSpan tolerance, string customMessage)
+        {
+            actual.AssertAwesomely(v => Is.Equal(v, expected, tolerance), actual, expected, tolerance, () => customMessage);
+        }
+
+        public static void ShouldBe(this DateTime actual, DateTime expected, TimeSpan tolerance, Func<string> customMessage)
```

Comment:
Could the other two overloads call this one, if we do internal refactoring later it just means less touch points if we call overloads rather than copy the code inside. i.e

```
 public static void ShouldBe(this DateTime actual, DateTime expected, TimeSpan tolerance, string customMessage)
{
    actual.ShouldBe(expected, tolerance, () => customMessage);
}
```